cdap-spark-core/src/main/scala/co/cask/cdap/app/runtime/spark/BatchReadableRDD.scala

```diff
@@ -37,6 +37,7 @@ import scala.reflect.ClassTag
   */
 class BatchReadableRDD[K: ClassTag, V: ClassTag](@transient sc: SparkContext,
                                                  @transient batchReadable: BatchReadable[K, V],
+                                                 datasetNamespace: Option[String],
```

Comment:
One question here. As the datasetNamespace can be empty actually(if user program don't provide this, it will get the dataset from the current namespace through `sprakDatasetContext`).  I think it won't be good to pack them into `DatasetId` with namespace field `null` right?